src/View/Helper/BreadcrumbsHelper.php

```diff
@@ -0,0 +1,318 @@
+
+            if (isset($separator['separator'])) {
+                $separatorParams['separator'] = $separator['separator'];
+                unset($separator['separator']);
+            }
+
+            if (isset($separator['templateVars'])) {
+                $separatorParams['templateVars'] = $separator['templateVars'];
+                unset($separator['templateVars']);
+            }
+
+            $separatorParams['attrs'] = $templater->formatAttributes($separator);
+        }
+
+        $crumbTrail = '';
+        foreach ($crumbs as $key => $crumb) {
+            $link = $this->prepareLink($crumb['link']);
+            $title = $crumb['title'];
+            $options = $crumb['options'];
+
+            $optionsLink = [];
+            if (isset($options['innerAttrs'])) {
+                $optionsLink = $options['innerAttrs'];
+                unset($options['innerAttrs']);
+            }
+
+            $template = 'item';
+            $templateParams = [
+                'attrs' => $templater->formatAttributes($options, ['templateVars']),
+                'innerAttrs' => $templater->formatAttributes($optionsLink),
+                'title' => $title,
+                'link' => $link,
+                'separator' => '',
+                'templateVars' => isset($options['templateVars']) ? $options['templateVars'] : []
+            ];
+
+            if (!($link)) {
+                $template = 'itemWithoutLink';
+            }
+
+            if ($separatorParams && $key !== ($crumbsCount - 1)) {
+                $templateParams['separator'] = $this->formatTemplate('separator', $separatorParams);
+            }
+
+            $crumbTrail .= $this->formatTemplate($template, $templateParams);
+        }
+
+        $crumbTrail = $this->formatTemplate('wrapper', [
+            'content' => $crumbTrail,
+            'attrs' => $templater->formatAttributes($attributes, ['templateVars']),
+            'templateVars' => isset($attributes['templateVars']) ? $attributes['templateVars'] : []
+        ]);
+
+        return $crumbTrail;
+    }
+
+    /**
+     * Search a crumb in the current stack which title matches the one provided as argument.
+     * If found, the index of the matching crumb will be returned.
+     *
+     * @param string $title Title to find
+     * @return int|null Index of the crumb found, or null if it can not be found
+     */
+    protected function findCrumb($title)
+    {
+        foreach ($this->crumbs as $key => $crumb) {
+            if ($crumb['title'] === $title) {
+                return $key;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Prepare the URL for a specific `link` param of a crumb
+     *
+     * @param array|string|null $link If array, an array of Router url params
+     * If string, will be used as is
+     * If empty, will consider that there is no link
+     *
+     * @return null|string The URL of a crumb
+     */
+    protected function prepareLink($link = null)
```

Comment:
No we can remove it. If the need to customize URL generation is brought to the table, we can still restore it when needed, it would not cause too much work.